Sources/Util/Heap.swift

```diff
@@ -64,13 +64,14 @@ class Heap<K: Comparable, V: Equatable>: Sequence, IteratorProtocol {
      * `delete` deletes the given value from this Heap.
      */
     func delete(_ node: HeapNode<K, V>) {
+        let lastIndexBeforeDeleted = self.nodes.count - 1
```

Comment:
@hackerwins If the parameter node of ```delete``` function is the last node, ```delete''' function does not work normally. Please, check it.

```
/**
   * `release` deletes the given value from this Heap.
   */
  public release(node: HeapNode<K, V>): void {
    const targetIndex = this.nodes.findIndex(
      (_node) => _node.getValue() === node.getValue(),
    );
    const lastNode = this.nodes.pop()!;

    if (targetIndex < 0 || !this.len()) {
      return;
    }

    this.nodes[targetIndex] = lastNode;

    this.heapify(this.getParentIndex(targetIndex), targetIndex);
  }
```